kentikapi/client.go

```diff
@@ -1,27 +1,41 @@
 package kentikapi
 
 import (
+	"context"
+	"crypto/tls"
+	"fmt"
 	"time"
 
+	grpcmiddleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	grpcretry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
+	grpccloudesxport "github.com/kentik/api-schema-public/gen/go/kentik/cloud_export/v202101beta1"
+	grpcsynthetics "github.com/kentik/api-schema-public/gen/go/kentik/synthetics/v202101beta1"
 	"github.com/kentik/community_sdk_golang/kentikapi/cloudexport"
 	"github.com/kentik/community_sdk_golang/kentikapi/internal/api_connection"
 	"github.com/kentik/community_sdk_golang/kentikapi/internal/httputil"
 	"github.com/kentik/community_sdk_golang/kentikapi/internal/resources"
 	"github.com/kentik/community_sdk_golang/kentikapi/synthetics"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
 )
 
 //nolint:gosec
 const (
 	authAPITokenKey = "X-CH-Auth-API-Token"
 	authEmailKey    = "X-CH-Auth-Email"
+	defaultTimeout  = 100 * time.Second
 )
 
 // Kentik API URLs.
 const (
-	APIURLUS          = "https://api.kentik.com/api/v5"
-	APIURLEU          = "https://api.kentik.eu/api/v5"
-	cloudExportAPIURL = "https://cloudexports.api.kentik.com"
-	syntheticsAPIURL  = "https://synthetics.api.kentik.com"
+	APIURLUS                = "https://api.kentik.com/api/v5"
+	APIURLEU                = "https://api.kentik.eu/api/v5"
+	cloudExportAPIURL       = "https://cloudexports.api.kentik.com"
+	syntheticsAPIURL        = "https://synthetics.api.kentik.com"
+	syntheticsGRPCHostPort  = "synthetics.api.kentik.com:443"
```

Comment:
We could parse/infer the host and port of the gRPC API from the `syntheticsAPIURL`. Due to the implementation of the service, it is 99.99% certain that the host and port will _always_ be the same for both HTTP/REST and gRPC APIs. It is safe to assume that if the `protocol` part of the URL is `https` (and the URL does not contain any explicit `port` specification) that the port is 443.

Reply:
cloudExportAPIURL and syntheticsAPIURL will be removed with HTTP API v6 code, so there will be no duplication.

We could tough construct syntheticsGRPCHostPort from APIURLUS, but that would hard-code `synthetics.` subdomain, which we should probably avoid.